envs/quadrotors/quad_multi_model.py

```diff
@@ -52,7 +52,49 @@ def forward(self, obs_dict):
         return out
 
 
-def register_models():
-    quad_custom_encoder_name = 'quad_multi_encoder_deepset'
+class QuadMultiHistogramEncoder(EncoderBase):
+    """Histogram encoder based on the DeepRL for Swarms Paper and Local Communication Protocols for Learning"""
+    def __init__(self, cfg, obs_space, timing, self_obs_dim=18, histogram_bins=64, histogram_hidden_size=64):
+        super().__init__(cfg, timing)
+        self.self_obs_dim = self_obs_dim
+        self.histogram_bins = histogram_bins
+        self.histogram_hidden_size = histogram_hidden_size
+
+        fc_encoder_layer = cfg.hidden_size
+        self.self_encoder = nn.Sequential(
+            nn.Linear(self.self_obs_dim, fc_encoder_layer),
+            nonlinearity(cfg),
+            nn.Linear(fc_encoder_layer, fc_encoder_layer),
+            nonlinearity(cfg)
+        )
+
+        self.histogram_encoder = nn.Sequential(
+            nn.Linear(self.histogram_bins, self.histogram_hidden_size),
+            nonlinearity(cfg),
+        )
```

Comment:
@alex-petrenko Correct me if I'm wrong, but I don't think we have separate encoders for self_observations and the histogram(?) The histogram embedding should be concatenated with self_observations and this should be fed into the policy_network that spits out actions.